stress: failed test in cockroach/server/status/status.test: TestMetricsRecorder

The following test appears to have failed:

```
=== RUN   TestMetricsRecorder
--- FAIL: TestMetricsRecorder (1.03s)
	recorder_test.go:287: recorder did not yield expected time series collection; diff:
		 [[6].Datapoints[0].Value: 10 != 0 [7].Datapoints[0].Value: 10 != 0 [8].Datapoints[0].Value: 10 != 0 [9].Datapoints[0].Value: 10 != 0 [10].Datapoints[0].Value: 10 != 0 [11].Datapoints[0].Value: 10 != 0 [12].Datapoints[0].Value: 10 != 0 [13].Datapoints[0].Value: 10 != 0 [48].Datapoints[0].Value: 10 != 0 [49].Datapoints[0].Value: 10 != 0 [50].Datapoints[0].Value: 10 != 0 [51].Datapoints[0].Value: 10 != 0 [52].Datapoints[0].Value: 10 != 0 [53].Datapoints[0].Value: 10 != 0 [54].Datapoints[0].Value: 10 != 0 [55].Datapoints[0].Value: 10 != 0 [96].Datapoints[0].Value: 10 != 0 [98].Datapoints[0].Value: 10 != 0 [100].Datapoints[0].Value: 10 != 0 [102].Datapoints[0].Value: 10 != 0 [104].Datapoints[0].Value: 10 != 0 [106].Datapoints[0].Value: 10 != 0 [108].Datapoints[0].Value: 10 != 0 [110].Datapoints[0].Value: 10 != 0]
```
Please assign, take a look and update the issue accordingly.

--- a/server/status/recorder_test.go
+++ b/server/status/recorder_test.go
@@ -166,6 +166,13 @@ func TestMetricsRecorder(t *testing.T) {
 	recorder.AddStore(store2)
 	recorder.NodeStarted(nodeDesc, 50)
 
+	// Ensure the metric system's view of time does not advance during this test
+	// as the test expects time to not advance too far which would age the actual
+	// data (e.g. in histogram's) unexpectedly.
+	defer metric.TestingSetNow(func() time.Time {
+		return time.Unix(0, manual.UnixNano()).UTC()
+	})()
+
 	// Create a flat array of registries, along with metadata for each, to help
 	// generate expected results.
 	regList := []struct {

--- a/util/metric/metric.go
+++ b/util/metric/metric.go
@@ -80,6 +80,16 @@ var _ periodic = &Rate{}
 
 var now = timeutil.Now
 
+// TestingSetNow changes the clock used by the metric system. For use by
+// testing to precisely control the clock.
+func TestingSetNow(f func() time.Time) func() {
+	origNow := now
+	now = f
+	return func() {
+		now = origNow
+	}
+}
+
 func maybeTick(m periodic) {
 	for m.nextTick().Before(now()) {
 		m.tick()

--- a/util/metric/metric_test.go
+++ b/util/metric/metric_test.go
@@ -59,6 +59,7 @@ func setNow(d time.Duration) {
 }
 
 func TestHistogramRotate(t *testing.T) {
+	defer TestingSetNow(nil)()
 	setNow(0)
 	h := NewHistogram(histWrapNum*time.Second, 1000+10*histWrapNum, 3)
 	var cur time.Duration
@@ -87,6 +88,8 @@ func TestHistogramRotate(t *testing.T) {
 }
 
 func TestHistogramJSON(t *testing.T) {
+	defer TestingSetNow(nil)()
+	setNow(0)
 	h := NewHistogram(0, 1, 3)
 	testMarshal(t, h, `[{"Quantile":100,"Count":0,"ValueAt":0}]`)
 	h.RecordValue(1)
@@ -94,6 +97,7 @@ func TestHistogramJSON(t *testing.T) {
 }
 
 func TestRateRotate(t *testing.T) {
+	defer TestingSetNow(nil)()
 	setNow(0)
 	const interval = 10 * time.Second
 	r := NewRate(interval)

